prov/sm2/src/sm2.h

```diff
@@ -176,11 +176,11 @@ static inline int sm2_match_tag(uint64_t tag, uint64_t ignore,
 static inline void sm2_generic_format(struct sm2_xfer_entry *xfer_entry,
 				      sm2_gid_t self_gid, uint32_t op,
 				      uint64_t tag, uint64_t cq_data,
-				      uint64_t op_flags, void *context)
+				      uint64_t *op_flags, void *context)
 {
 	xfer_entry->hdr.op = op;
 	/* We only care about lower 32 bits */
-	xfer_entry->hdr.op_flags = (uint32_t) op_flags;
+	xfer_entry->hdr.op_flags = (uint32_t) *op_flags;
 	xfer_entry->hdr.tag = tag;
 	xfer_entry->hdr.sender_gid = self_gid;
 	xfer_entry->hdr.cq_data = cq_data;
```

Comment:
I don't understand this commit's purpose.  It says change flags to pointer to allow modification, but it doesn't modify them.

Also, why would generic format do such a modification?  It seems like if we allow this, it should be protocol-specific to decide to do so, not the job of generic_format.

Can we remove this commit?